lib/video/screenRecording/firefox/firefoxWindowRecorder.js

```diff
@@ -146,6 +171,8 @@ module.exports = class FirefoxWindowRecorder {
         // Nothing to see here
       }
     }
+    this.recordingStartTime = findRecordingStartTime(this.baseDir);
```

Comment:
Each of these invocations enumerates all files in the directory.  So does `generateVideo` itself.  Figure out how to do the directory walk once.  You might make `generateVideo` return (or set) these additional values; or you might extract the "walk directory, collect file information" function and send the results into `generateVideo`.  Your call.